lib/assign-properties.js

```diff
@@ -54,16 +62,35 @@ if (sourceAvailable) {
   }
 }
 
+function mergeDescriptors(descriptor, targetDescriptor) {
+  if (descriptor.hasOwnProperty('value') || !targetDescriptor) {
+    return descriptor;
+  }
+
+  if ( !hasGetter(descriptor) && hasGetter(targetDescriptor) ) {
+    descriptor.get = targetDescriptor.get;
+  }
+
+  if ( !hasSetter(descriptor) && hasSetter(targetDescriptor) ) {
+    descriptor.set = targetDescriptor.set;
+  }
+
+  return descriptor;
+}
+
 function assignProperties(target, options) {
-  var value;
+  var value, descriptor, targetDescriptor;
 
   for (var key in options) {
-    value = options[key];
+    descriptor = Object.getOwnPropertyDescriptor(options, key);
+    value = descriptor.value;
 
     if (typeof value === 'function' && hasSuper(value)) {
       target[key] = giveMethodSuper(getPrototypeOf(target), key, value);
     } else {
-      target[key] = value;
+      targetDescriptor = Object.getOwnPropertyDescriptor(target.__proto__, key);
```

Comment:
i would love for us to use (or use a polyfil of) `Object.getOwnPropertyDescriptors`